packages/node_modules/@webex/plugin-meetings/src/meeting-info/utilv2.js

```diff
@@ -0,0 +1,226 @@
+  const sipUri = DIALER_REGEX.SIP_ADDRESS.exec(sipString);
+
+  return sipUri;
+};
+
+MeetingInfoUtil.isPhoneNumber = (phoneNumber) => {
+  const isValidNumber = DIALER_REGEX.PHONE_NUMBER.test(phoneNumber);
+
+  return isValidNumber;
+};
+
+MeetingInfoUtil.getHydraId = (destination) => {
+  const {type, id, cluster} = deconstructHydraId(destination);
+
+  if (id && UUID_REG.test(id)) {
+    if (type === _ROOM_) {
+      return {room: true, destination: id, cluster};
+    }
+    if (type === _PEOPLE_) {
+      return {people: true, destination: id, cluster};
+    }
+
+    return {};
+  }
+
+  return {};
+};
+
+MeetingInfoUtil.getSipUriFromHydraPersonId = (destination, webex) => webex.people.get(destination).then((res) => {
+  if (res.emails && res.emails.length) {
+    return res.emails[0];
+  }
+  throw new ParameterError('Hydra Id Lookup was an invalid hydra person id.');
+}).catch((err) => {
+  LoggerProxy.logger.error(`Meeting-info:util#MeetingInfoUtil.getSipUriFromHydraPersonId --> getSipUriFromHydraPersonId ${err} `);
+  throw err;
+});
+
+
+MeetingInfoUtil.getDestinationType = async (from) => {
+  const {destination, type, webex} = from;
+
+  if (type) {
+    return {
+      destination,
+      type
+    };
+  }
+  const options = {};
+  const hydraId = MeetingInfoUtil.getHydraId(destination);
+
+  if (MeetingInfoUtil.isMeetingLink(destination)) {
+    LoggerProxy.logger.warn('Meeting-info:util#generateOptions --> WARN, use of Meeting Link is deprecated, please use a SIP URI instead');
+
+    options.type = _MEETING_LINK_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isSipUri(destination)) {
+    options.type = _SIP_URI_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isPhoneNumber(destination)) {
+    options.type = _SIP_URI_;
+    options.destination = destination;
+  }
+  else if (MeetingInfoUtil.isConversationUrl(destination, webex)) {
+    options.type = _CONVERSATION_URL_;
+    options.destination = destination;
+  }
+  else if (hydraId.people) {
+    options.type = _SIP_URI_;
+
+    return MeetingInfoUtil.getSipUriFromHydraPersonId(hydraId.destination, webex).then((res) => {
+      options.destination = res;
+
+      // Since hydra person ids require a unique case in which they are
+      // entirely converted to a SIP URI, we need to set a flag for detecting
+      // this type of destination.
+      options.wasHydraPerson = true;
+
+      return Promise.resolve(options);
+    });
+  }
+  else if (hydraId.room) {
+    options.type = _CONVERSATION_URL_;
+    try {
+      await webex.internal.services.waitForCatalog('postauth');
+
+      const conversationUrl = webex.internal.conversation.getUrlFromClusterId({
+        cluster: hydraId.cluster,
+        id: hydraId.destination
+      });
+
+      options.destination = conversationUrl;
+    }
+    catch (e) {
+      LoggerProxy.logger.error(`Meeting-info:util#generateOptions --> ${e}`);
```

Comment:
wrong function name in the log, should be getDestinationType

Reply:
done